### PR TITLE
feat: BOL updates

### PR DESCRIPTION
**Pull request by nissimsan**
Added various common BOL elements:
- Shipping parties
- Carrier scac code
- Origin and destination locations

### CODE REVIEW

File: packages/traceability-schemas/schemas/BillOfLading.json [hunk 1]
```diff
@@ -20,20 +20,62 @@
       ]
     },
     "billOfLadingNumber": {
-      "$comment": "{\"term\": \"billOfLadingNumber\", \"@id\": \"https://schema.org/identifier\"}",
+      "$comment": "{\"term\": \"billOfLadingNumber\", \"@id\": \"https://service.unece.org/trade/uncefact/vocabulary/uncl1153/#Bill_of_lading_number\"}",
```
Comment (mkhraisha) [inline packages/traceability-schemas/schemas/BillOfLading.json:23]: Not sure why this is added?

this appears to be a Buy Ship Pay Reference Data Model  vocabulary, not entirely sure this is appropriate for the generic BillOfLading

**Review by mkhraisha — COMMENTED** [review]
I'm not very familiar with the new vocabulary source, it appears more specific to ship based transports.
Also some fields from Freight would probably be duplicated on some root level fields unless I'm mistaken.

File: packages/traceability-schemas/schemas/BillOfLading.json [hunk 2]
```diff
@@ -20,20 +20,62 @@
...
       "title": "Bill Of Lading Number",
-      "description": "This identifeir might not be globally unique.",
+      "description": "This identifier might not be globally unique.",
       "type": "string"
     },
+    "bookingNumber":{
+      "$comment": "{\"term\": \"billOfLadingNumber\", \"@id\": \"https://service.unece.org/trade/uncefact/vocabulary/uncl1153/#Consignment_identifier_carrier_assigned\"}",
```
Comment (mkhraisha) [inline packages/traceability-schemas/schemas/BillOfLading.json:29]: Booking number -> Bill of ladingnumber

File: packages/traceability-schemas/schemas/BillOfLading.json [hunk 3]
```diff
@@ -20,20 +20,62 @@
...
-      "title": "Purchase ",
+      "title": "Purchase",
       "description": "Purchase documents related to a bill of lading",
       "type": "array",
       "items": {
         "$ref": "https://w3id.org/traceability/schemas/Purchase.json"
       }
     },
+    "scac": {
+      "$comment": "{\"term\": \"freightForwarder\", \"@id\": \"https://service.unece.org/trade/uncefact/vocabulary/uncl1153/#Standard_Carrier_Alpha_Code_(SCAC)_number\"}",
```
Comment (mkhraisha) [inline packages/traceability-schemas/schemas/BillOfLading.json:44]: freightforwarder -> scac

File: packages/traceability-schemas/schemas/BillOfLading.json [hunk 4]
```diff
@@ -20,20 +20,62 @@
...
+      "title": "SCAC",
+      "description": "For maritime shipments, this code qualifies a Standard Alpha Carrier Code (SCAC) as issued by the United Stated National Motor Traffic Association Inc.",
+      "type": "string"
+    }, 
+    "carrier": {
+      "$comment": "{\"term\": \"freightForwarder\", \"@id\": \"https://service.unece.org/trade/uncefact/vocabulary/uncefact/#carrierParty\"}",
```
Comment (mkhraisha) [inline packages/traceability-schemas/schemas/BillOfLading.json:50]: freightforwarder -> carrier

File: packages/traceability-schemas/schemas/BillOfLading.json (hunk 1 above)
Comment (nissimsan) [reply packages/traceability-schemas/schemas/BillOfLading.json]: @mkhraisha, sticking with schema.org/identifier. I might raise a separate PR/issue sometime dedicated to continue the discussion.

File: packages/traceability-schemas/schemas/BillOfLading.json (hunk 3 above)
Comment (nissimsan) [reply packages/traceability-schemas/schemas/BillOfLading.json]: *facepalm* - thanks!

File: packages/traceability-schemas/schemas/BillOfLading.json (hunk 4 above)
Comment (nissimsan) [reply packages/traceability-schemas/schemas/BillOfLading.json]: *facepalm* - thanks!

**Comment by nissimsan** [comment]
@mkhraisha, regarding lastExit and firstArrival that you mentioned yesterday, I looked at `ParcelDelivery` and those aren't the same. BOLs operate with four locations: 
1. `placeOfReceipt`
2. `portOfLoading`
3. `portOfDischarge`
4. `placeOfDelivery`
`ParcelDelivery` has the two "outermost" locations (first mile and last mile, if you will). These ones represent the "innermost" loading and discharge ports. 
Here's an example - all carriers have this: 
image

File: packages/traceability-schemas/schemas/BillOfLading.json (hunk 2 above)
Comment (nissimsan) [reply packages/traceability-schemas/schemas/BillOfLading.json]: bookingNb != billOfLadingNb

**Review by OR13 — APPROVED** [review]

**Review by mprorock — APPROVED** [review]

**Pull request merged**